core/src/main/java/com/linecorp/armeria/client/WebClientRequestPreparation.java

```diff
@@ -213,6 +215,98 @@ public <T> FutureTransformingRequestPreparation<ResponseEntity<T>> asJson(Class<
+     * the default {@link ObjectMapper}.
+     * {@link HttpStatus} type argument specify what type of response is allowed.
+     * For example:
+     * <pre>{@code
+     * WebClient client = WebClient.of("https://api.example.com");
+     * CompletableFuture<ResponseEntity<MyObject>> response =
+     *     client.prepare()
+     *           .get("/v1/items/1")
+     *           .asJson(MyObject.class, HttpStatus.INTERNAL_SERVER_ERROR)
+     *           .execute();
+     * }</pre>
+     *
+     * <p>Note that this method should NOT be used if the result type is a container such as
+     * {@link Collection} or {@link Map}. Use {@link #asJson(TypeReference, HttpStatus)} for the container type.
+     *
+     * @throws InvalidHttpResponseException if the {@link HttpStatus} of the response is not
+     *                                      same to {@link HttpStatus} type argument or fails to decode
+     *                                      the response body into the result type.
+     * @see JacksonObjectMapperProvider
+     */
+    @UnstableApi
+    public <T> FutureTransformingRequestPreparation<ResponseEntity<T>> asJson(Class<? extends T> clazz,
+                                                                              HttpStatus httpStatus) {
+        requireNonNull(httpStatus, "httpStatus");
+        return asJson(clazz, new HttpStatusPredicate(httpStatus));
+    }
+
+    /**
+     * Deserializes the JSON content of the {@link HttpResponse} into the specified non-container type using
+     * the default {@link ObjectMapper}.
+     * {@link HttpStatusClass} type argument specify what type of response is allowed.
+     * For example:
+     * <pre>{@code
+     * WebClient client = WebClient.of("https://api.example.com");
+     * CompletableFuture<ResponseEntity<MyObject>> response =
+     *     client.prepare()
+     *           .get("/v1/items/1")
+     *           .asJson(MyObject.class, HttpStatusClass.SERVER_ERROR)
+     *           .execute();
+     * }</pre>
+     *
+     * <p>Note that this method should NOT be used if the result type is a container such as
+     * {@link Collection} or {@link Map}.
+     * Use {@link #asJson(TypeReference, HttpStatusClass)} for the container type.
+     *
+     * @throws InvalidHttpResponseException if the {@link HttpStatusClass} of the response is not
+     *                                      same to {@link HttpStatusClass} type argument or fails to decode
+     *                                      the response body into the result type.
+     * @see JacksonObjectMapperProvider
+     */
+    @UnstableApi
+    public <T> FutureTransformingRequestPreparation<ResponseEntity<T>> asJson(Class<? extends T> clazz,
+                                                                              HttpStatusClass httpStatusClass) {
+        requireNonNull(httpStatusClass, "httpStatusClass");
+        return asJson(clazz, new HttpStatusClassPredicate(httpStatusClass));
```

Comment:
`HttpStatusClass` is an enum so we can statistically create `HttpStatusClassPredicate`s.
How about adding a static factory method for `HttpStatusClassPredicate`?
```java
// Returns a pre-populated object.
HttpStatusClassPredicate.of(httpStatusClass)
```
We may apply a similar approach to `HttpStatusPrediate`

Reply:
If we apply a static factory method for `HttpStatusPredicate`, the number of static variables is as same as HttpStatus.
Conditional branching of static methods also increases as well.
If so, should we apply it?